File: Example/JFMinimalNotification/ViewController.m
//
//  ViewController.m
//  JFMinimalNotification
//
//  Created by Jeremy Fox on 5/4/13.
//  Copyright (c) 2013 Jeremy Fox. All rights reserved.
//

#import "ViewController.h"
#import "JFMinimalNotification.h"

@interface ViewController () <JFMinimalNotificationDelegate, UITextFieldDelegate>
@property (nonatomic, strong) JFMinimalNotification* minimalNotification;
@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.titleLabelTextField.text = @"Testing";
    self.subTitleLabelTextField.text = @"This is my awesome sub-title";
    
    /**
     * Create the notification
     */
    self.minimalNotification = [JFMinimalNotification notificationWithStyle:JFMinimalNotificationStyleBlurLight title:self.titleLabelTextField.text subTitle:self.subTitleLabelTextField.text dismissalDelay:0.0 touchHandler:^{
        [self.minimalNotification dismiss];
    }];
    
    self.minimalNotification.edgePadding = UIEdgeInsetsMake(0, 0, 10, 0);
    
    [self.view addSubview:self.minimalNotification];
    
    /**
     * Set the delegate
     */
    self.minimalNotification.delegate = self;
    
    /**
     * Set the desired font for the title and sub-title labels
     * Default is System Normal
     */
    UIFont* titleFont = [UIFont fontWithName:@"STHeitiK-Light" size:22];
    [self.minimalNotification setTitleFont:titleFont];
    UIFont* subTitleFont = [UIFont fontWithName:@"STHeitiK-Light" size:16];
    [self.minimalNotification setSubTitleFont:subTitleFont];
    
    /**
     * Uncomment the following line to present notifications from the top of the screen.
     */
    // self.minimalNotification.presentFromTop = YES;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

- (void)showToastWithMessage:(NSString *)message {
    if (self.minimalNotification) {
        [self.minimalNotification dismiss];
        [self.minimalNotification removeFromSuperview];
        self.minimalNotification = nil;
    }
    
    self.minimalNotification = [JFMinimalNotification notificationWithStyle:JFMinimalNotificationStyleError
                                                          title:NSLocalizedString(@"Refresh Error", @"Refresh Error")
                                                       subTitle:message
                                                 dismissalDelay:10.0];
    
    /**
     * Set the desired font for the title and sub-title labels
     * Default is System Normal
     */
    UIFont* titleFont = [UIFont systemFontOfSize:22.0];
    [self.minimalNotification setTitleFont:titleFont];
    UIFont* subTitleFont = [UIFont systemFontOfSize:16.0];
    [self.minimalNotification setSubTitleFont:subTitleFont];
    
    /**
     * Add the notification to a view
     */
    [self.view addSubview:self.minimalNotification];
    
    // show
    [self performSelector:@selector(showNotification) withObject:nil afterDelay:0.1];
}

- (void)showNotification {
    [self.minimalNotification show];
}

- (IBAction)show:(id)sender {
    [self.minimalNotification show];
}

- (IBAction)dismiss:(id)sender {
    [self.minimalNotification dismiss];
}

- (IBAction)setLightStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleBlurLight animated:YES];
}

- (IBAction)setDarkStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleBlurDark animated:YES];
}

- (IBAction)setCustomStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleCustom animated:YES];
}

- (IBAction)setErrorStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleError animated:YES];
}

- (IBAction)setSuccessStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleSuccess animated:YES];
}

- (IBAction)setDefaultStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleDefault animated:YES];
}

- (IBAction)setInfoStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleInfo animated:YES];
}

- (IBAction)setWarningStyle:(id)sender {
    [self.minimalNotification setStyle:JFMinimalNotificationStyleWarning animated:YES];
}

- (IBAction)setLeftView:(id)sender {
    /**
     * Set the left view with any UIView or UIView subclass.
     */
    [self.minimalNotification setLeftAccessoryView:[[UIImageView alloc] initWithImage:[UIImage imageNamed:@"thumbs-up.jpg"]] animated:YES];
}

- (IBAction)setRightView:(id)sender {
    /**
     * Set the right view with any UIView or UIView subclass
     * This will automatically recalculate the title and sub-title frame and animate to the new position
     */
    [self.minimalNotification setRightAccessoryView:[[UIImageView alloc] initWithImage:[UIImage imageNamed:@"thumbs-up.jpg"]] animated:YES];
}

- (IBAction)removeLeftView:(id)sender {
    /**
     * Remove an existing left view by passing in nil
     * This will automatically recalculate the title and sub-title frame and animate to the new position
     */
    [self.minimalNotification setLeftAccessoryView:nil animated:YES];
}

- (IBAction)removeRightView:(id)sender {
    /**
     * Remove an existing right view by passing in nil
     * This will automatically recalculate the title and sub-title frame and animate to the new position
     */
    [self.minimalNotification setRightAccessoryView:nil animated:YES];
}

#pragma mark ----------------------
#pragma mark UITextFieldDelegate
#pragma mark ----------------------

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    JFMinimalNotificationStyle style = self.minimalNotification.currentStyle;
    [self.minimalNotification removeFromSuperview];
    self.minimalNotification = nil;
    self.minimalNotification = [JFMinimalNotification notificationWithStyle:style title:self.titleLabelTextField.text subTitle:self.subTitleLabelTextField.text dismissalDelay:0.0f touchHandler:^{
        [self.minimalNotification dismiss];
    }];
    self.minimalNotification.delegate = self;
    UIFont* titleFont = [UIFont fontWithName:@"STHeitiK-Light" size:22];
    [self.minimalNotification setTitleFont:titleFont];
    UIFont* subTitleFont = [UIFont fontWithName:@"STHeitiK-Light" size:16];
    [self.minimalNotification setSubTitleFont:subTitleFont];
    [self.view addSubview:self.minimalNotification];
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        [self.minimalNotification show];
    });
    
    return YES;
}

#pragma mark ----------------------
#pragma mark JFMinimalNotificationDelegate
#pragma mark ----------------------

- (void)minimalNotificationWillShowNotification:(JFMinimalNotification*)notification {
    NSLog(@"willShowNotification");
}

- (void)minimalNotificationDidShowNotification:(JFMinimalNotification*)notification {
    NSLog(@"didShowNotification");
}

- (void)minimalNotificationWillDisimissNotification:(JFMinimalNotification*)notification {
    NSLog(@"willDisimissNotification");
}

- (void)minimalNotificationDidDismissNotification:(JFMinimalNotification*)notification {
    NSLog(@"didDismissNotification");
}

@end

File: Example/JFMinimalNotification/ViewController.swift
//
//  ViewController.swift
//  JFMinimalNotification
//
//  Created by Jeremy E Fox on 9/26/24.
//  Copyright © 2024 Jeremy Fox. All rights reserved.
//

import UIKit

class ViewController: UIViewController, JFMinimalNotificationDelegate, UITextFieldDelegate {
    
    var minimalNotification: JFMinimalNotification?
    
    @IBOutlet weak var titleLabelTextField: UITextField!
    @IBOutlet weak var subTitleLabelTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        titleLabelTextField.text = "Testing"
        subTitleLabelTextField.text = "This is my awesome sub-title"
        
        // Create the notification
        minimalNotification = JFMinimalNotification(
            withStyle: .custom,
            title: titleLabelTextField.text ?? "",
            subTitle: subTitleLabelTextField.text ?? ""
        ) {
            self.minimalNotification?.dismiss()
        }
        
        minimalNotification?.edgePadding = UIEdgeInsets(top: 0, left: 0, bottom: 10, right: 0)
        view.addSubview(minimalNotification!)
        minimalNotification?.backgroundColor = .purple
        minimalNotification?.titleLabel?.textColor = .white
        minimalNotification?.subTitleLabel?.textColor = .white
        
        // Set the delegate
        minimalNotification?.delegate = self
        
        // Set the desired font for title and subtitle labels
        if let titleFont = UIFont(name: "STHeitiK-Light", size: 22) {
            minimalNotification?.setTitleFont(titleFont)
        }
        if let subTitleFont = UIFont(name: "STHeitiK-Light", size: 16) {
            minimalNotification?.setSubTitleFont(subTitleFont)
        }
        
        // Uncomment to present notification from the top of the screen
//         minimalNotification?.presentFromTop = true
    }
    
    func showToast(withMessage message: String) {
        minimalNotification?.dismiss()
        minimalNotification?.removeFromSuperview()
        minimalNotification = nil
        
        minimalNotification = JFMinimalNotification(
            withStyle: .error,
            title: NSLocalizedString("Refresh Error", comment: "Refresh Error"),
            subTitle: message,
            dismissalDelay: 10.0
        )
        
        // Set the desired font for title and subtitle labels
        minimalNotification?.setTitleFont(UIFont.systemFont(ofSize: 22.0))
        minimalNotification?.setSubTitleFont(UIFont.systemFont(ofSize: 16.0))
        
        // Add the notification to a view
        view.addSubview(minimalNotification!)
        
        // Show after a delay
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
            self.showNotification()
        }
    }
    
    @objc func showNotification() {
        minimalNotification?.show()
    }
    
    @IBAction func show(_ sender: Any? = nil) {
        minimalNotification?.show()
    }
    
    @IBAction func dismiss(_ sender: Any? = nil) {
        minimalNotification?.dismiss()
    }
    
    @IBAction func setLightStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.blurLight, animated: true)
    }
    
    @IBAction func setDarkStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.blurDark, animated: true)
    }
    
    @IBAction func setCustomStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.custom, animated: true)
    }
    
    @IBAction func setErrorStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.error, animated: true)
    }
    
    @IBAction func setSuccessStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.success, animated: true)
    }
    
    @IBAction func setDefaultStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.default, animated: true)
    }
    
    @IBAction func setInfoStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.info, animated: true)
    }
    
    @IBAction func setWarningStyle(_ sender: Any? = nil) {
        minimalNotification?.setStyle(.warning, animated: true)
    }
    
    @IBAction func setLeftView(_ sender: Any? = nil) {
        // Set the left view with any UIView or UIView subclass
        minimalNotification?.setLeftAccessoryView(UIImageView(image: UIImage(named: "thumbs-up.jpg")), animated: true)
    }
    
    @IBAction func setRightView(_ sender: Any? = nil) {
        // Set the right view with any UIView or UIView subclass
        minimalNotification?.setRightAccessoryView(UIImageView(image: UIImage(named: "thumbs-up.jpg")), animated: true)
    }
    
    @IBAction func removeLeftView(_ sender: Any? = nil) {
        // Remove an existing left view by passing nil
        minimalNotification?.setLeftAccessoryView(nil, animated: true)
    }
    
    @IBAction func removeRightView(_ sender: Any? = nil) {
        // Remove an existing right view by passing nil
        minimalNotification?.setRightAccessoryView(nil, animated: true)
    }
    
    // MARK: - UITextFieldDelegate
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        let style = minimalNotification?.currentStyle
        minimalNotification?.removeFromSuperview()
        minimalNotification = nil
        minimalNotification = JFMinimalNotification(
            withStyle: style ?? .custom,
            title: titleLabelTextField.text ?? "",
            subTitle: subTitleLabelTextField.text ?? "",
            dismissalDelay: 0.0
        ) {
            self.minimalNotification?.dismiss()
        }
        minimalNotification?.delegate = self
        minimalNotification?.setTitleFont(UIFont(name: "STHeitiK-Light", size: 22)!)
        minimalNotification?.setSubTitleFont(UIFont(name: "STHeitiK-Light", size: 16)!)
        view.addSubview(minimalNotification!)
        
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
            self.minimalNotification?.show()
        }
        
        return true
    }
    
    // MARK: - JFMinimalNotificationDelegate
    
    func minimalNotificationWillShow(notification: JFMinimalNotification) {
        print("willShowNotification")
    }
    
    func minimalNotificationDidShow(notification: JFMinimalNotification) {
        print("didShowNotification")
    }
    
    func minimalNotificationWillDismiss(notification: JFMinimalNotification) {
        print("willDisimissNotification")
    }
    
    func minimalNotificationDidDismiss(notification: JFMinimalNotification) {
        print("didDismissNotification")
    }
}
